frontend/src/repo-menu.tsx

```diff
@@ -0,0 +1,59 @@
+import { useAllPulls, useSetFilter } from './pulldasher/pulls-context';
+import { useArrayUrlState } from './use-url-state';
+import { Button, Menu, MenuButton, MenuList, MenuOptionGroup, MenuItemOption } from "@chakra-ui/react";
+import { useEffect, useCallback, useMemo } from 'react'
+
+export function RepoMenu() {
+   const pulls = useAllPulls();
+   const [selectedRepos, setSelectedRepos] = useArrayUrlState('repo', []);
+   // Nothing selected == show everything, otherwise, it'd be empty
+   const showAll = selectedRepos.length === 0;
+   const setPullFilter = useSetFilter();
+
+   const allRepos = useMemo(() => Array.from(new Set(pulls.map((pull) => pull.getRepoName()))), [pulls]);
+   const selectedReposSet = useMemo(() => new Set(selectedRepos), [selectedRepos]);
+
+   const onSelectedChange = useCallback((newSelectedRepos: string | string[]) => {
+      newSelectedRepos = Array.from(newSelectedRepos);
+      // If they've selected *all* the repos, then let's reset it to the
+      // default (empty array == show all repos)
+      if (newSelectedRepos.length == allRepos.length) {
+         newSelectedRepos = [];
+      }
+
+      // Update the url
+      setSelectedRepos(newSelectedRepos);
+
+      // Update the pull filter
+      const selectedReposSet = new Set(newSelectedRepos);
+      setPullFilter('repo', selectedReposSet.size === 0 ? null : (pull) =>
+         selectedReposSet.has(pull.getRepoName())
+      );
```

Comment:
I'm betting the fact that there are two places with similar state like this is going to bite us eventually. We may want to move toward allowing the pull filters to be based on the URL state or something down the road.

Redux may be worth considering for some of this, because it provides a global location to store state which can be accessed from multiple places.

Reply:
My first reaction was "Yes, url is the state, that's how it was done". So I looked more closely at the code and you are totally right, that's *not* how it's currently done and there's a bug: history navigation is broken!

Fixing!

Reply:
Fixed.